src/main/database.cpp

```diff
@@ -85,13 +87,25 @@ std::unique_ptr<storage::BufferManager> Database::initBufferManager(const Databa
         db.dbConfig.bufferPoolSize, db.dbConfig.maxDBSize, db.vfs.get(), db.dbConfig.readOnly);
 }
 
+std::string expandPath(main::ClientContext* context, const std::string& path) {
```

Comment:
I think this case would be handled by the `isSubdirectory` function to convert to a abusolute path without .. and . 

I added the convert in the expandPath as well just in case